libbeat/outputs/elasticsearch/output.go

```diff
@@ -29,41 +40,21 @@ var (
 	ErrResponseRead = errors.New("bulk item status parse failed.")
 )
 
-const (
-	defaultMaxRetries = 3
-
-	defaultBulkSize = 50
-
-	elasticsearchDefaultTimeout = 90 * time.Second
-)
-
 func init() {
 	outputs.RegisterOutputPlugin("elasticsearch", elasticsearchOutputPlugin{})
 }
 
-type elasticsearchOutputPlugin struct{}
-
-type elasticsearchOutput struct {
-	index string
-	mode  mode.ConnectionMode
-
-	topology
-}
-
 // NewOutput instantiates a new output plugin instance publishing to elasticsearch.
 func (f elasticsearchOutputPlugin) NewOutput(
-	config *outputs.MothershipConfig,
+	cfg *ucfg.Config,
 	topologyExpire int,
 ) (outputs.Outputer, error) {
-
-	// configure bulk size in config in case it is not set
-	if config.BulkMaxSize == nil {
-		bulkSize := defaultBulkSize
-		config.BulkMaxSize = &bulkSize
+	if !cfg.HasField("bulk_max_size") {
```

Comment:
default is 2048, but elasticsearch sets default to 50 if not configured. The bulk-collector is part of publisher pipeline and must be updated in config if not set.